mmr/peaksold.go

```diff
@@ -0,0 +1,60 @@
+package mmr
+
+// Note: the expectation is that once we are satisfied with the new methods we
+// will delete this file A reason to keep it around is that testing may benefit
+// from having multiple implementations of key algorithms
+
+// PeaksOld is deprecated and retained only for reference and testing.
+//
+// returns the array of mountain peaks in the MMR. This is completely
+// deterministic given a valid mmr size. If the mmr size is invalid, this
+// function returns nil.
+//
+// It is guaranteed that the peaks are listed in ascending order of position
+// value.  The highest peak has the lowest position and is listed first. This is
+// a consequence of the fact that the 'little' 'down range' peaks can only appear
+// to the 'right' of the first perfect peak, and so on recursively.
+//
+// Note that as a matter of implementation convenience and efficency the peaks
+// are returned as *one based positions*
+//
+// So given the example below, which has an mmrSize of 17, the peaks are [15, 18]
+//
+//	3            15
+//	           /    \
+//	          /      \
+//	         /        \
+//	2       7          14
+//	      /   \       /   \
+//	1    3     6    10     13      18
+//	    / \  /  \   / \   /  \    /  \
+//	0  1   2 4   5 8   9 11   12 16   17
+func PeaksOld(mmrSize uint64) []uint64 {
```

Comment:
I think we can make this more managable, either by completely removing old code (we have it released and versioned) or moving it to its own module that is better named than old?

Reply:
The plan is to delete everything called 'old' as soon as the updates are stable in prod.

For the "Bagged" support, it is possible we can get rid of that too but it would require a migration of existing seals. This is less "hard" than it sounds but its still something that could easily make replicated logs look "broken" even though the are provably good.